### PR TITLE
Fix #36: Prevent Invalid System/AuxiliaryCode Calls

### LINKED ISSUE #36: Prevent users from making calls to system-level classes.
Students should not be able to access AuxiliaryCode or System classes.

Repro steps (at least, the easiest ones):

Navigate to Problem 2
Paste in the following code:
`def isBalanced(s):
    return AuxiliaryCode.countNumberOfParentheses(s)
`
Run code. You should get an error message (rather than a SyntaxError or something) since it called one of the AuxiliaryCode methods. We should block students from calling any of those classes, potentially via throwing a SyntaxError or a ForbiddenNamespaceError of our own making. Something to tell them yo you can't run our code.

### PR DESCRIPTION
**Pull request by dchen97**
Fix to Issue #36 
* Added 2 PrereqCheckFailure types to account for invalid calls to the `System` and `AuxiliaryCode` classes
* Added new feedback for these new PrereqCheckFailure types in `FeedbackGenerator`
* Added relevant tests

### CODE REVIEW

**Comment by dchen97** [comment]
Had an idea about making this more sustainable. Will work on it more so it's not ready for review yet.

**Comment by dchen97** [comment]
OK I've made some adjustments but I'm not sure they're actually all that effective or helpful in terms of scalability for how things are right now. I'm open to suggestions! Otherwise, this should be ready for review now.

File: client/app.js [hunk 1]
```diff
@@ -122,6 +122,20 @@ tie.constant('SYSTEM_CODE', {
...
 });
 
+/**
+ * Provides class names for system and auxiliary code according to the language
+ * that is being used.
+ *
+ * @type {{}}
+ * @constant
+ */
+tie.constant('SYSTEM_AUXILIARY_CLASSES', {
```
Comment (seanlip) [inline client/app.js]: I thought we already had constants for these (cf one of Elena's recent PRs). I think it's also fine for them to be consistent across all languages?

File: client/services/FeedbackGeneratorService.js [hunk 2]
```diff
@@ -413,6 +413,26 @@ tie.factory('FeedbackGeneratorService', [
...
             '-- we cannot process code in the global scope.'
           ].join(' '));
+        } else if (prereqCheckFailure.hasInvalidAuxiliaryCodeCall()) {
+          feedback.appendTextParagraph([
+            'Looks like your code had a runtime error. Here is the error ',
+            'message: '
+          ].join(''));
+          feedback.appendCodeParagraph([
+            'ForbiddenNamespaceError: It looks like you\'re trying to call ',
+            'the AuxiliaryCode class or its methods, which is forbidden. ',
```
Comment (seanlip) [inline client/services/FeedbackGeneratorService.js]: Error message should make use of the constant in app.js.

File: client/services/FeedbackGeneratorService.js [hunk 3]
```diff
@@ -413,6 +413,26 @@ tie.factory('FeedbackGeneratorService', [
...
+            'Please resubmit without using this class.'
+          ].join(''));
+        } else if (prereqCheckFailure.hasInvalidSystemCall()) {
+          feedback.appendTextParagraph([
+            'Looks like your code had a runtime error. Here is the error ',
+            'message: '
+          ].join(''));
+          feedback.appendCodeParagraph([
+            'ForbiddenNamespaceError: It looks you\'re using the System class ',
+            'or its methods, which is forbidden. Please resubmit without ',
```
Comment (seanlip) [inline client/services/FeedbackGeneratorService.js]: Ditto.

File: client/services/code_evaluators/PythonPrereqCheckService.js [hunk 4]
```diff
@@ -164,6 +192,14 @@ tie.factory('PythonPrereqCheckService', [
             PREREQ_CHECK_TYPE_GLOBAL_CODE, null, starterCode);
         }
 
+        if (checkInvalidSystemClassCalls(code) === SYSTEM_CALL) {
```
Comment (seanlip) [inline client/services/code_evaluators/PythonPrereqCheckService.js]: Would expect something more like: `if (hasInvalidSystemClassCalls(code))`, and `if (hasInvalidAuxiliaryClassCalls(code))`. Then you don't need to pass those strings around.

File: client/app.js (hunk 1 above)
Comment (dchen97) [inline client/app.js]: Done!

File: client/services/FeedbackGeneratorService.js (hunk 2 above)
Comment (dchen97) [inline client/services/FeedbackGeneratorService.js]: Done!

File: client/services/FeedbackGeneratorService.js (hunk 3 above)
Comment (dchen97) [inline client/services/FeedbackGeneratorService.js]: Done!

**Pull request merged**